src/mavoscript.js

```diff
@@ -321,7 +321,62 @@ var _ = Mavo.Script = {
 				return ret;
 			},
 			precedence: 3
-		}
+		},
+		"groupby": {
+			symbol: "by",
+			code: (array, key) => {
+				if (Array.isArray(array) && Array.isArray(key))  { 
+					var propName = key[Mavo.as] === true ? key[Mavo.name] : $.value(key[0], Mavo.toNode, "property");
+					var temp = new Mavo.BucketMap({arrays: true});
+					var ret = [];
+					ret[Mavo.groupedBy] = true;
+
+					for (var i = 0; i < array.length; i++) {
+						const k = i < key.length ? Mavo.value(key[i]) : null;
+						temp.set(k, Mavo.value(array[i]));
+					}	
+
+					for (var [value, items] of temp) {
+						var obj = {"$value": value};
+
+						if (propName !== undefined) {
+							obj[propName] = value;
+						}
+						
+						obj["$items"] = items;
+						ret.push(obj);
+					}
+
+					return ret;
+				}
+				else { // invalid arguments, convert to arrays
+					// return (Mavo.toArray(array), Mavo.toArray(key));
+				}
+			},
+			precedence: 2
+		},
+		"as": {
+			symbol: "as",
+			code: (property, name) => {
+				if (property !== undefined && $.type(property) === "array" && name !== undefined) {
+					if ($.type(name) === "string" || $.value(name[0], Mavo.toNode, "property") !== undefined) {
```

Comment:
You're checking if `name` is a string. What if `name` actually corrsponds to a property?
E.g. `foo as bar` and you already have a `bar` property?
IMO it should still work, and just extract the property from the value, via `Mavo.property`. Thoughts?

Reply:
I agree that it should work in some form but I see two different options. 
Say `bar` is a property set to `"hello"`
(1) `foo as bar` would set `Mavo.as = "bar"`
(2) `foo as bar` would set `Mavo.as = "hello"`

I am implementing (1) in this commit and think it is the way to go but wanted to put both options out there and get your feedback. The issue I have with (2) is that I think most likely when someone uses `foo as bar` they don't realize that they have a property `bar` or are intending to set the value to `"bar"`. It seems a little confusing to me if it goes and gets named the value of `bar` given that `bar` could be of any type.

Reply:
I agree it should be 1, but excellent observation!